### ISSUE TITLE
Anonymous classes and onlyDefs

### ISSUE THREAD

**Issue by wmdietl** [typetools/checker-framework#2227]
This small example produces an error from the Nullness Checker:

```java
import org.checkerframework.checker.nullness.qual.Nullable;
class Test {
  @Nullable Object f;
  Runnable r = new Runnable() {
      public void run() {
          f.toString();
      }
  };
}
```

However, there is no way to use `-AonlyDefs` to check this class including all contained anonymous classes.

`-AonlyDefs=Test` should check a class plus all contained anonymous classes (and lambdas).
The current check for the class name in `onlyDefs` should be extended to also check the enclosing top-level class.
This check is in https://github.com/typetools/checker-framework/blob/master/framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java method `boolean shouldSkipDefs(ClassTree node)`

This was raised in https://github.com/typetools/checker-framework/issues/904#issuecomment-445409730 . It is very useful to be able to select the scope of type checking, but then very surprising if anonymous classes aren't checked.

**Comment by wmdietl** [comment]
@NazguL86 Do you want to take a stab at fixing this?
